src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -129,25 +127,29 @@ public HikariPool(final HikariConfig config)
 
       LinkedBlockingQueue<Runnable> addQueue = new LinkedBlockingQueue<>(config.getMaximumPoolSize());
       this.addConnectionQueue = unmodifiableCollection(addQueue);
-      this.addConnectionExecutor = createThreadPoolExecutor(addQueue, poolName + " connection adder", threadFactory, new ThreadPoolExecutor.DiscardPolicy());
-      this.closeConnectionExecutor = createThreadPoolExecutor(config.getMaximumPoolSize(), poolName + " connection closer", threadFactory, new ThreadPoolExecutor.CallerRunsPolicy());
+      this.addConnectionExecutor = createAddConnectionExecutor(threadFactory, addQueue);
+      this.closeConnectionExecutor = createThreadPoolExecutor(config.getMaximumPoolSize(), poolName + " connection " +
+            "closer",
+         threadFactory, new ThreadPoolExecutor.CallerRunsPolicy());
 
       this.leakTaskFactory = new ProxyLeakTaskFactory(config.getLeakDetectionThreshold(), houseKeepingExecutorService);
 
-      this.houseKeeperTask = houseKeepingExecutorService.scheduleWithFixedDelay(new HouseKeeper(), 100L, housekeepingPeriodMs, MILLISECONDS);
-
-      if (Boolean.getBoolean("com.zaxxer.hikari.blockUntilFilled") && config.getInitializationFailTimeout() > 1) {
-         addConnectionExecutor.setCorePoolSize(Runtime.getRuntime().availableProcessors());
-         addConnectionExecutor.setMaximumPoolSize(Runtime.getRuntime().availableProcessors());
+      this.houseKeeperTask = houseKeepingExecutorService.scheduleWithFixedDelay(new HouseKeeper(), 100L,
+         housekeepingPeriodMs, MILLISECONDS);
+   }
 
-         final long startTime = currentTime();
-         while (elapsedMillis(startTime) < config.getInitializationFailTimeout() && getTotalConnections() < config.getMinimumIdle()) {
-            quietlySleep(MILLISECONDS.toMillis(100));
-         }
+   private ThreadPoolExecutor createAddConnectionExecutor(ThreadFactory threadFactory,
+                                                          LinkedBlockingQueue<Runnable> addQueue)
+   {
+      int poosSize = 1;
 
-         addConnectionExecutor.setCorePoolSize(1);
-         addConnectionExecutor.setMaximumPoolSize(1);
+      if (Boolean.getBoolean("com.zaxxer.hikari.blockUntilFilled")
+         && config.getInitializationFailTimeout() > 1) {
+         poosSize = config.getMinimumIdle();
       }
+
+      return createThreadPoolExecutorWithFixedSize(addQueue, poolName + " connection adder",
+         threadFactory, new ThreadPoolExecutor.DiscardPolicy(), poosSize);
    }
```

Comment:
This code substantially changes the behavior of HikariCP.

Currently, HikariCP starts up with a single-threaded "connection adder".  If "com.zaxxer.hikari.blockUntilFilled" is specified, it **_temporarily_** sets the "connection adder" thread pool size to ``availableProcessors()``.  But **_after_** the pool is filled, it returns the thread pool size back to ``1`` (single-threaded).

HikariCP is not designed for multiple "connection adder" threads _while the pool is in use_.  We allow multiple adder threads only here, during initialization, because we know that the pool will not be accessed during this time.

The proposed code does not return the pool size back to ``1`` after the pool is initialized.

However, the most serious issue with this proposed change is that it does not actually "block" the initialization at all when "_com.zaxxer.hikari.blockUntilFilled_" is requested!  The whole point of this flag is to block the initialization thread in the constructor until the pool reaches ``minimumIdle`` connections.

What is going on here?  It makes me suspect that you do not actually understand the code.

Reply:
Sorry for the mistake in my previous code, your suspect makes me clearer of situation now.

I don't see the trigger of pool initialization (starting the `addConnectionExecutor` to work) in origin code of this place (actually it is scheduled by the `houseKeeperTask`) , so I treat the `while` loop as a mistake and remove it from the code., which is not a good behavior, sorry again for that. But origin code is apparently wrong if `blockUntilFilled` is true, because the pool executor is always created with a max pool size of 1, change it to available processor num (or `MinimumIdle`) will trigger a `IllegalArgumentException`.

After changing again, all my code did now is just as the pull request's title says.

As for the format and typo, I will treat them more carefully.

Any further suggestions will be highly appreciated.